container_test.go

```diff
@@ -57,22 +57,22 @@ func TestRegister(t *testing.T) {
 	tts := []struct {
 		name  string
 		param interface{}
-		err   error
+		err   string
 	}{
-		{"non function", struct{}{}, errParamType},
-		{"wrong return count", noReturn, errReturnCount},
-		{"non pointer return", returnNonPointer, errReturnKind},
-		{"wrong parameters type", nonPointerParams, errArgKind},
-		{"pointer Provide", &struct{}{}, nil},
+		{"non function", struct{}{}, "for constructor struct {}: " + errParamType.Error()},
+		{"wrong return count", noReturn, "for constructor func(): " + errReturnCount.Error()},
+		{"non pointer return", returnNonPointer, "for constructor func() dig.S: " + errReturnKind.Error()},
```

Comment:
no, that is not the intention. The intention was to identify what func call is the cause. e.g., knowing "func() dig.S" returns no pointer is more valuable.